tf2onnx/convert.py

```diff
@@ -193,56 +228,230 @@ def main():
+            large_model=large_model,
+            tensors_to_rename=tensors_to_rename,
+            initialized_tables=initialized_tables,
+            output_path=output_path)
+
+        return model_proto, external_tensor_storage
+
+
+def from_function(function, input_signature=None, opset=None, custom_ops=None, custom_op_handlers=None,
+                  custom_rewriter=None, inputs_as_nchw=None, extra_opset=None, shape_override=None, target=None,
+                  large_model=False, output_path=None):
+    """Returns a ONNX model_proto for a tf.function.
+
+    Args:
+        function: the tf.function we want to convert
+        input_signature: a tf.TensorSpec or a numpy array defining the shape/dtype of the input
+        opset: the opset to be used for the ONNX model, default is the latest
+        target: list of workarounds applied to help certain platforms
+        custom_op_handlers: dictionary of custom ops handlers
+        custom_rewriter: list of custom graph rewriters
+        extra_opset: list of extra opset's, for example the opset's used by custom ops
+        shape_override: dict with inputs that override the shapes given by tensorflow
+        inputs_as_nchw: transpose inputs in list from nchw to nhwc
+        large_model: use the ONNX external tensor storage format
+        output_path: save model to output_path
+
+    Returns:
+        An ONNX model_proto and an external_tensor_storage dict.
+    """
+    if LooseVersion(tf.__version__) < "2.0":
+        raise NotImplementedError("from_keras requires tf-1.15 or newer")
+
+    concrete_func = function.get_concrete_function(*input_signature)
+
+    input_names = [input_tensor.name for input_tensor in concrete_func.inputs
+                   if input_tensor.dtype != tf.dtypes.resource]
+    output_names = [output_tensor.name for output_tensor in concrete_func.outputs
+                    if output_tensor.dtype != tf.dtypes.resource]
+
+    initialized_tables = None
+    tensors_to_rename = tensor_names_from_structed(concrete_func, input_names, output_names)
+
+    with tf.device("/cpu:0"):
+        frozen_graph = tf_loader.from_function(concrete_func, input_names, output_names)
+        model_proto, external_tensor_storage = _convert_common(
+            frozen_graph,
+            name=concrete_func.name,
+            continue_on_error=True,
+            target=None,
+            opset=opset,
+            custom_op_handlers=custom_ops,
+            extra_opset=extra_opset,
+            shape_override=shape_override,
+            input_names=input_names,
+            output_names=output_names,
+            inputs_as_nchw=inputs_as_nchw,
+            large_model=large_model,
+            tensors_to_rename=tensors_to_rename,
+            initialized_tables=initialized_tables,
+            output_path=output_path)
+
+        return model_proto, external_tensor_storage
+
+
+def from_graph(graph_def, name=None, input_names=None, output_names=None, opset=None, custom_ops=None,
```

Comment:
```suggestion
def from_graph_def(graph_def, name=None, input_names=None, output_names=None, opset=None, custom_ops=None,
```

The input is a graph_def not a tf graph

Reply:
done